### PR TITLE
Markets-Index processes error handling

### PR DESCRIPTION
**Pull request by semyonoskin**

### CODE REVIEW

File: modules/markets-index/src/main/scala/org/ergoplatform/dex/index/processes/PoolsIndexing.scala [hunk 1]
```diff
@@ -62,12 +67,21 @@ object PoolsIndexing {
         val assets        = poolSnapshots.flatMap(p => List(p.entity.lp.id, p.entity.x.id, p.entity.y.id)).distinct
         val commit        = batch.lastOption.map(_.commit)
 
+        val retryPolicy = limitRetries[F](30) |+| constantDelay[F](10.seconds)
```
Comment (GusevTimofey) [inline modules/markets-index/src/main/scala/org/ergoplatform/dex/index/processes/PoolsIndexing.scala:70]: values to config

File: modules/markets-index/src/main/scala/org/ergoplatform/dex/index/processes/PoolsIndexing.scala [hunk 2]
```diff
@@ -62,12 +67,21 @@ object PoolsIndexing {
...
               txr.trans(NonEmptyList.fromList(assets).fold(List.empty[TokenId].pure[D])(repos.assets.existing))
             unknownAssets = assets.diff(existingAssets)
-            assetsInfo <- unknownAssets.filterNot(_ == ErgoAssetId).traverse(explorer.getTokenInfo)
+            assetsInfo <- unknownAssets
+                            .filterNot(_ == ErgoAssetId)
+                            .flatTraverse(tknId =>
+                              retryingOnFailures(
+                                retryPolicy,
+                                (opt: Option[TokenInfo]) => opt.isDefined,
+                                (_: Option[TokenInfo], _: RetryDetails) => ().pure[F]
```
Comment (GusevTimofey) [inline modules/markets-index/src/main/scala/org/ergoplatform/dex/index/processes/PoolsIndexing.scala:82]: log retry

File: modules/markets-index/src/main/scala/org/ergoplatform/dex/index/processes/PoolsIndexing.scala (hunk 1 above)
Comment (GusevTimofey) [inline modules/markets-index/src/main/scala/org/ergoplatform/dex/index/processes/PoolsIndexing.scala:70]: Point to discussion @oskin1:

In all sql queries we think, that token info exists for pool in db. But the situation when we failed to get token info in the end of our retry policy is possible. 

We have some ways to resolve this issue:
1. Simply drop pool for token without info. We lose some information but it's an analytics services. 
2. Poll explorer forever (and investigate by hands if the poll will be really infinite).
3. The dangerest - insert pool without token info.

**Review by GusevTimofey — CHANGES_REQUESTED** [review]

File: modules/markets-index/src/main/scala/org/ergoplatform/dex/index/processes/PoolsIndexing.scala (hunk 1 above)
Comment (oskin1) [reply modules/markets-index/src/main/scala/org/ergoplatform/dex/index/processes/PoolsIndexing.scala]: Normally token info must be present for any pool in explorer db. Usually the error here is caused by non 2xx/4xx response codes (response timeout usually I guess). In this case retry is effectively a solution. In the case of 4xx code we have an inconsistent explorer view and the service should halt indexing and wait for manual investigation.

File: modules/markets-index/src/main/scala/org/ergoplatform/dex/index/processes/PoolsIndexing.scala [hunk 3]
```diff
@@ -62,12 +73,21 @@ object PoolsIndexing {
...
             unknownAssets = assets.diff(existingAssets)
-            assetsInfo <- unknownAssets.filterNot(_ == ErgoAssetId).traverse(explorer.getTokenInfo)
+            assetsInfo <- unknownAssets
+                            .filterNot(_ == ErgoAssetId)
+                            .flatTraverse(tknId =>
+                              retryingOnFailures(
+                                retryPolicy,
+                                (opt: Option[TokenInfo]) => opt.isDefined,
+                                (_: Option[TokenInfo], _: RetryDetails) => info"Failed to find token $tknId. Retrying..."
+                              )(explorer.getTokenInfo(tknId)).map(_.toList)
```
Comment (oskin1) [inline modules/markets-index/src/main/scala/org/ergoplatform/dex/index/processes/PoolsIndexing.scala:89]: Retry makes sense only in the case of non-2xx/4xx response code. In the case of 4xx code indexer should panic.

**Review by oskin1 — APPROVED** [review]

**Pull request merged**